src/panels/config/energy/dialogs/dialog-energy-gas-settings.ts

```diff
@@ -82,15 +90,19 @@ export class DialogEnergyGasSettings
     }
 
     const pickableUnit =
-      this._pickableUnit ||
-      (this._params.allowedGasUnitClass === undefined
-        ? "ft³, m³, Wh, kWh, MWh or GJ"
+      this._params.allowedGasUnitClass === undefined
+        ? [...(this._gas_units || []), ...(this._energy_units || [])].join(", ")
         : this._params.allowedGasUnitClass === "energy"
-        ? "Wh, kWh, MWh or GJ"
-        : "ft³ or m³");
+        ? this._energy_units?.join(", ") || ""
+        : this._gas_units?.join(", ") || "";
+
+    const unitPrice = this._pickedDisplayUnit
+      ? `${this.hass.config.currency}/${this._pickedDisplayUnit}`
+      : undefined;
 
     const externalSource =
-      this._source.stat_cost && isExternalStatistic(this._source.stat_cost);
+      this._source.stat_energy_from &&
+      isExternalStatistic(this._source.stat_energy_from);
```

Comment:
We should check if the gas statistic is external, not if the cost statistic is external.